examples/xxz_dmrg.jl

```diff
@@ -59,13 +63,16 @@ H = xxz_chain(sites; cutoff=1e-9);
 # format, namely as a *matrix product state* (MPS). This is achieved by `ITensors.dmrg`
 # which is wrapped in the [`DMRG`](@ref) solver type:
 
-dm = DMRG(; n_states=1, tol_degeneracy=0.0, sweeps=default_sweeps(; cutoff_max=1e-9),
+dm = DMRG(; sweeps=default_sweeps(; cutoff_max=1e-9),
           observer=() -> DMRGObserver(; energy_tol=1e-9));
 
-# While the implemented DMRG solver is capable of solving degenerate ground states, we here
-# opt for non-degenerate settings, i.e. `n_states=1` and `tol_degeneracy=0.0`. (We do this
-# due to a ``L+1``-fold degeneracy on the parameter domain, where the degenerate DMRG solver
-# can produce instable results for larger ``L``.)
+# For each solve a new `ITensors.DMRGObserver` object is created that converges the DMRG 
+# iteration up the specified `energy_tol`. The sweeps argument takes any `ITensors.Sweeps`
+# object that controls the approximation settings with respect to the DMRG sweeps.
+# While the implemented DMRG solver is capable of also solving degenerate ground states,
+# we here opt for the default settings, which perform a non-degenerate DMRG solve (We do
+# this due to a ``L+1``-fold degeneracy on the parameter domain, where the degenerate DMRG
+# solver can produce instable results for larger ``L``.)
```

Comment:
I don't get this as it's written. Why do we use the default observer? Because of the degeneracies or because we don't care about degeneracies?

Reply:
So the observer is independent of whether we care about degeneracies or not. In this case the default keyword arguments for `n_states` and `tol_degeneracy` lead to a non-degenerate solve, which I tried to explain here. I will make that more explicit in the next commit.

I could also completely omit the `sweeps` and `observer` keyword arguments since the default values are taken, but I wanted to explain some of the differences that occur when using the `DMRG` solver.

Reply:
I understand this, but it's really hard to grasp from what you wrote. With your changes it becomes better, but still it is a bit tricky to read.